Student/StudentUnitTests/Assignments/AssignmentList/AssignmentListViewControllerTests.swift

```diff
@@ -0,0 +1,216 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2019-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import UIKit
+import XCTest
+@testable import Student
+@testable import Core
+import TestsFoundation
+
+class AssignmentListViewControllerTests: PersistenceTestCase {
+
+    var vc: AssignmentListViewController!
+    let courseID = "1"
+    let baseURL = URL(string: "https://canvas.instructure.com/")!
+    var req: AssignmentListRequestable!
+    var gradingPeriods: [APIAssignmentListGradingPeriod] = []
+    var groups: [APIAssignmentListGroup] = []
+
+    override func setUp() {
+        vc = AssignmentListViewController.create(env: env, courseID: courseID)
+        gradingPeriods = []
+        groups = []
+    }
+
+    func mockNetwork() {
+        api.mock(GetCustomColorsRequest(), value: APICustomColors(custom_colors: [ "course_\(courseID)": "#008EE2", ]))
+        api.mock(GetCourseRequest(courseID: courseID), value: .make())
+        api.mock(req, data: data(gradingPeriods: gradingPeriods, groups: groups) )
+    }
+
+    func data(gradingPeriods: [APIAssignmentListGradingPeriod], groups: [APIAssignmentListGroup]) -> Data {
+        let encoder = JSONEncoder()
+        encoder.dateEncodingStrategy = .iso8601
+        let gpData = try! encoder.encode(gradingPeriods)
+        let grpData = try! encoder.encode(groups)
+        let strGrdPeriod = String(data: gpData, encoding: .utf8)!
+        let strGrp = String(data: grpData, encoding: .utf8)!
+        print("json: \(strGrdPeriod)")
+
+        let str = """
+        {
+         "data": {
+           "course": {
+             "name": "CourseName",
+             "gradingPeriods": {
+               "nodes": \(strGrdPeriod)
+             },
+             "groups": {
+               "nodes": \(strGrp)
+             }
+           }
+         }
+        }
+        """
```

Comment:
Instead of this string could you encode a `APIAssignmentListResponse` and convert that to a string?